src/operations/utils.js

```diff
@@ -0,0 +1,111 @@
+import bitcoinjs from 'bitcoinjs-lib'
+import RIPEMD160 from 'ripemd160'
+import bigi from 'bigi'
+
+export const DUST_MINIMUM = 5500
+
+export function hash160(buff: Buffer) {
+  const sha256 = bitcoinjs.crypto.sha256(buff)
+  return (new RIPEMD160()).update(sha256).digest()
+}
+
+export function hash128(buff: Buffer) {
+  return Buffer.from(bitcoinjs.crypto.sha256(buff).slice(0, 16))
+}
+
+
+// COPIED FROM coinselect, because 1 byte matters sometimes.
+// baseline estimates, used to improve performance
+const TX_EMPTY_SIZE = 4 + 1 + 1 + 4
+const TX_INPUT_BASE = 32 + 4 + 1 + 4
+const TX_INPUT_PUBKEYHASH = 107
+const TX_OUTPUT_BASE = 8 + 1
+const TX_OUTPUT_PUBKEYHASH = 25
+
+function inputBytes(input) {
+  return TX_INPUT_BASE + (input.script && input.script.length > 0 ?
+                          input.script.length : TX_INPUT_PUBKEYHASH)
+}
+
+function outputBytes(output) {
+  return TX_OUTPUT_BASE + (output.script ? output.script.length : TX_OUTPUT_PUBKEYHASH)
+}
+
+function transactionBytes(inputs, outputs) {
+  return TX_EMPTY_SIZE +
+    inputs.reduce((a, x) => (a + inputBytes(x)), 0) +
+    outputs.reduce((a, x) => (a + outputBytes(x)), 0)
+}
+
+//
+
+export function estimateTXBytes(txIn : bitcoinjs.Transaction | bitcoinjs.TransactionBuilder,
```

Comment:
`txIn` type is listed as `bitcoinjs.Transaction` but then later on the code checks if `txIn` is a `bitcoinjs.TransactionBuilder`. does this function take both types?

Reply:
yep -- it takes both types, I think that the `|` there indicates that it can take either type.